examples/spreadinterponly1d.cpp

```diff
@@ -0,0 +1,91 @@
+// this is all you must include for the finufft lib...
+#include <finufft.h>
+
+// also used in this example...
+#include <cassert>
+#include <chrono>
+#include <complex>
+#include <cstdio>
+#include <stdlib.h>
+#include <vector>
+using namespace std;
+using namespace std::chrono;
+
+int main(int argc, char *argv[])
+/* Example of double-prec spread/interp only tasks, with basic math tests.
+   Complex I/O arrays, but recall the kernel is real.  Barnett 1/8/25.
+
+   The math tests are:
+   1) for spread, check sum of spread kernel masses is as expected from sum
+   of strengths (ie testing the zero-frequency component in NUFFT).
+   2) for interp, check each interp kernel mass is the same as from one.
+
+   Without knowing the kernel, this is about all that can be done!
+   (Better math tests would be, ironically, to wrap the spreader/interpolator
+   into a NUFFT and test that :) But we already have that in FINUFFT.)
+
+   Compile and run (static library case):
+
+   g++ spreadinterponly1d.cpp -I../include ../lib-static/libfinufft.a -o
+   spreadinterponly1d -lfftw3 -lfftw3_omp && ./spreadinterponly1d
+
+   See: spreadtestnd for usage of internal (non FINUFFT-API) spread/interp.
+*/
+{
+  int M = 1e7; // number of nonuniform points
+  int N = 1e7; // size of regular grid
+  finufft_opts opts;
+  finufft_default_opts(&opts);
+  opts.spreadinterponly = 1;    // task: the following two control kernel used...
+  double tol            = 1e-9; // tolerance for (real) kernel shape design only
+  opts.upsampfac        = 2.0;  // pretend upsampling factor (really no upsampling)
+  // opts.spread_kerevalmeth = 0;  // would be needed for any nonstd upsampfac
+
+  complex<double> I = complex<double>(0.0, 1.0); // the imaginary unit
+  vector<double> x(M);                           // input
+  vector<complex<double>> c(M);                  // input
+  vector<complex<double>> F(N);                  // output (spread to this array)
+
+  // first spread M=1 single unit-strength at the origin, only to get its total mass...
+  x[0]       = 0.0;
+  c[0]       = 1.0;
+  int unused = 1;
+  int ier    = finufft1d1(1, &x[0], &c[0], unused, tol, N, &F[0], &opts); // warm-up
```

Comment:
x.data(), c.data(), F.data()

&x[0] is `hacky` we should not encourage users to do so.